pkg/reconciler/taskrun/resources/taskref.go

```diff
@@ -181,43 +180,48 @@ func resolveTask(ctx context.Context, resolver remote.Resolver, name string, kin
 }
 
 // readRuntimeObjectAsTask tries to convert a generic runtime.Object
-// into a v1beta1.TaskObject type so that its meta and spec fields
+// into a *v1.Task type so that its meta and spec fields
 // can be read. v1 object will be converted to v1beta1 and returned.
 // An error is returned if the given object is not a
 // TaskObject or if there is an error validating or upgrading an
 // older TaskObject into its v1beta1 equivalent.
 // TODO(#5541): convert v1beta1 obj to v1 once we use v1 as the stored version
-func readRuntimeObjectAsTask(ctx context.Context, obj runtime.Object) (v1beta1.TaskObject, error) {
+func readRuntimeObjectAsTask(ctx context.Context, obj runtime.Object) (*v1.Task, error) {
 	switch obj := obj.(type) {
-	case v1beta1.TaskObject:
-		return obj, nil
-	case *v1.Task:
-		t := &v1beta1.Task{
+	case *v1beta1.ClusterTask:
+		//TODO
+	case *v1beta1.Task:
+		// TODO: confirm that this shall be deprecated?
+		t := &v1.Task{
 			TypeMeta: metav1.TypeMeta{
 				Kind:       "Task",
-				APIVersion: "tekton.dev/v1beta1",
+				APIVersion: "tekton.dev/v1",
 			},
 		}
-		if err := t.ConvertFrom(ctx, obj); err != nil {
+		if err := obj.ConvertTo(ctx, t); err != nil {
 			return nil, err
 		}
 		return t, nil
+
+	case *v1.Task:
+		return obj, nil
 	}
 	return nil, errors.New("resource is not a task")
 }
 
 // LocalTaskRefResolver uses the current cluster to resolve a task reference.
 type LocalTaskRefResolver struct {
 	Namespace    string
-	Kind         v1beta1.TaskKind
+	Kind         v1.TaskKind
 	Tektonclient clientset.Interface
 }
 
 // GetTask will resolve either a Task or ClusterTask from the local cluster using a versioned Tekton client. It will
 // return an error if it can't find an appropriate Task for any reason.
 // TODO: if we want to set source for in-cluster task, set it here.
 // https://github.com/tektoncd/pipeline/issues/5522
-func (l *LocalTaskRefResolver) GetTask(ctx context.Context, name string) (v1beta1.TaskObject, *v1beta1.ConfigSource, error) {
+func (l *LocalTaskRefResolver) GetTask(ctx context.Context, name string) (*v1.Task, *v1.ConfigSource, error) {
+	// TODO to figure out whether we can delete the ClusterTaskKind check since we are only having ClusterTasks in v1beta1
 	if l.Kind == v1beta1.ClusterTaskKind {
```

Comment:
I am not sure how to deal with the ClusterTaskKind condition check once we changed #L215 to v1.TaskKind since ClusterTasks could have duplicate names with Tasks.